### PR TITLE
[NTH-90] create a route that get user data

### PR DESCRIPTION
**Pull request by raghdarizq**
based on his token (image, name, role, userID, therapistId)

### CODE REVIEW

File: src/routes/auth.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import { Router } from 'express';
-import { login } from '../controllers';
+import { login, getAuth } from '../controllers';
+import { checkToken } from '../middlewares';
 
 const router = Router();
 router.post('/login', login);
+router.get('/login', checkToken(), getAuth);
```
Comment (LinaYahya) [inline src/routes/auth.ts:7]:
```suggestion
router.get('/', checkToken(), getAuth);
```

File: src/middlewares/checkToken.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { Response, NextFunction } from 'express';
+import { templateErrors } from '../helpers';
+import { verifyToken } from '../helpers/jwt';
+import { Decode, RequestWithUserRole } from '../types';
+
+const checkTokenForLogin = () => async (
```
Comment (LinaYahya) [inline src/middlewares/checkToken.ts:6]:
```suggestion
const checkTokenForLogin = async (
```
No need to declare a function that return express middleware, we did that in checkAuth cause we need to call it with different roles
 ```js
 checkAuth('therapist');
checkAuth('user');
```
and checkAuth was a function that return a middleware, But here we can do the follow
```js
const checkToken = (req, res, next) => {}
// and call it 
route(checkToken, ourController)
```

Comment (LinaYahya) [inline src/middlewares/checkToken.ts:6]: please rename it

Comment (LinaYahya) [inline src/middlewares/checkToken.ts:6]: can we move this to same file of checkauth?

File: src/controllers/auth.ts [hunk 3]
```diff
@@ -49,5 +52,49 @@ const login = async (req: Request, res: Response, next: NextFunction) => {
     next(err);
   }
 };
+const getAuth = async (
+  req: RequestWithUserRole,
+  res: Response,
+  next: NextFunction,
+) => {
+  let data;
+  if (req.user?.role === 'therapist') {
```
Comment (LinaYahya) [inline src/controllers/auth.ts:61]:
```suggestion
  if (req.user?.role === RolesForSelected.therapist) {
```

File: src/controllers/auth.ts [hunk 4]
```diff
@@ -49,5 +52,49 @@ const login = async (req: Request, res: Response, next: NextFunction) => {
...
+  } else {
+    const admin = await Admin.findOne({
+      attributes: ['username', 'id'],
+      where: {
+        id: req.user?.userId,
+      },
+    });
+    data = admin;
+  }
+  try {
```
Comment (LinaYahya) [inline src/controllers/auth.ts:90]: why not wrapping the whole controller with try, and please move to a service

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: src/middlewares/checkauth.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { Response, NextFunction } from 'express';
-import { templateErrors, verifyToken } from '../helpers';
-
+import { templateErrors } from '../helpers';
+import { verifyToken } from '../helpers/jwt';
```
Comment (LinaYahya) [inline src/middlewares/checkauth.ts:3]: please import from helpers index file

**Review by LinaYahya — APPROVED** [review]

**Review by MohannadSabaa — APPROVED** [review]

**Review by YasserAmro0 — APPROVED** [review]

**Review by ahmedmurtaja — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**